### DISCUSSION TITLE
environmental variables to pass by nodejs to react-app

### DISCUSSION THREAD

**Discussion by nikonov91-dev** [JamesIves/github-pages-deploy-action#611] [Q&A] [answered]
Hi there,
I cannot find the mention of **ENV** key in *yml configuration* in [your docs](https://github.com/JamesIves/github-pages-deploy-action#configuration-).

Is it correct information explained in [this answer of stackoverflow](https://stackoverflow.com/a/59320508/9277453)?
Because they reference to you but when I configure in that way environmental variables  and expect to access them from debugger console *process.env.TEST* - they are undefined.

![image](https://user-images.githubusercontent.com/31586394/108559863-94e48600-7304-11eb-9fa4-74fa4fec280c.png)

but I cannot access any of them from debugger in devtool.

![image](https://user-images.githubusercontent.com/31586394/108560030-ccebc900-7304-11eb-9289-510b2df5d898.png)

Thanks

**Comment by JamesIves** [comment, marked as answer]
It looks like what you're trying to do is modify your pages build script based on an environment variable, the build step that occurs before the deployment isn't handled by this action. 

In the Chrome console `process.env` will be undefined as the environment variables are only defined within the workflow, once your page is built and deployed they are no longer accessible. 

This question would be better suited for StackOverflow as it doesn't really pertain to this project. If you're interested in learning more about how environment variables work and why they are used in some projects I'd [recommend checking out this article](https://medium.com/chingu/an-introduction-to-environment-variables-and-how-to-use-them-f602f66d15fa).
